app/models/points.js

```diff
@@ -13,10 +15,28 @@ class Points extends Collection {
 
   // Optional: Validation function
   static validate(doc) {
-    if (!doc.title || !doc.description) {
-      return { error: 'Title and description are required' }
+    Joi.objectId = JoiObjectID(Joi)
+
+    const Integer = /^[0-9]+$/
+    const ObjectID = /^[0-9a-fA-F]{24}$/
+    const SANE = 4096;
+
+    const String = () => Joi.string().allow('').max(SANE);
+    const IsoDate = () => Joi.string().allow('').isoDate();
+    const Email = () => Joi.string().allow('').email();
+
+    const pointSchema = Joi.object({
```

Comment:
pointSchema needs to be something that can be used by the api functions to test the data that is coming in.
Something like this:
export pointSchema={
      _id: Joi.objectId()
      title: String(),
      description: String(),
      demInfo: demInfo(),
     parentId: ObjectID,
}

and then in the api, like in app/api/upsert-why.js
 
it can do someting like
```
import {pointSchema} from ' ../model/points.js'

//and then it can do
 ['_id','title','description','parentId'].forEach(prop=>{
   if(!pointObj[prop]) { handle fail }
   if(!pointSchema[prop]) { handle fail}
  // not sure how to use joi in this way but something like
   if(pointSchema[prop](pointObj[prop]).error) {handle errro}
}
```
The object is that at the api interface, the api knows what fields are required and needs to make much tougher tests on the data - (like length) - than the model needs to make.

We also need a test in the api for properties that are not defined in the schema, whereas in the model we don't have to be so concerned.